packages/tldraw/src/lib/shapes/shared/useEditableText.ts

```diff
@@ -207,3 +236,46 @@ export function useEditableText<T extends Extract<TLShape, { props: { text: stri
 		isEmpty,
 	}
 }
+
+function getCaretPosition(editor: Editor, inputEl: HTMLTextAreaElement | null) {
```

Comment:
instead of all the manual geometry & text measurement here, we could use some browser-native APIs:
```ts
// browser doesn't let us measure ranges inputs for some reason, but we have
// the `text-content` el which we can use. probs safer to pass this as a ref tho
const measureEl = inputEl.previousSibling

// there should only be one child - the text node 
const measureNode = measureEl.childNodes[0]

// the range API is awkward but lets u do cool stuff:
const range = document.createRange()
range.setStart(measureNode, inputEl.selectionStart)
range.setEnd(measureNode, inputEl.selectionEnd)
const rangeBounds = range.getBoundingClientRect()

// lets say cursor position is bottom-center
const cursorScreenPos = new Vec(rangeBounds.bottom, rangeBounds.left + rangeBounds.width / 2)
const cursorViewportPos = Vec.Add(editor.getViewportScreenBounds(), cursorScreenPos)

return cursorViewportPos

Reply:
haha, that's funny - that _was_ my first version of how I wrote this (i'm very familiar with the range API from before heh). i was trying to think ahead to this review and thought you might all ask me to not rewrite logic that existed in `editor.textMeasure.measureText` 😛 

again, especially after offline conversation of making the text labels more swappable in general, the more general logic fits that paradigm much better — i'll rework to use the general range API.